bundle/src/main/java/com/adobe/acs/commons/remoteassets/impl/RemoteAssetsConfigImpl.java

```diff
@@ -0,0 +1,343 @@
+        return this.username;
+    }
+
+    /**
+     * @see RemoteAssetsConfig#getPassword()
+     * @return String
+     */
+    @Override
+    public String getPassword() {
+        return this.password;
+    }
+
+    /**
+     * @see RemoteAssetsConfig#getTagSyncPaths()
+     * @return List<String>
+     */
+    @Override
+    public List<String> getTagSyncPaths() {
+        return this.tagSyncPaths;
+    }
+
+    /**
+     * @see RemoteAssetsConfig#getDamSyncPaths()
+     * @return List<String>
+     */
+    @Override
+    public List<String> getDamSyncPaths() {
+        return this.damSyncPaths;
+    }
+
+    /**
+     * @see RemoteAssetsConfig#getRetryDelay()
+     * @return Integer
+     */
+    @Override
+    public Integer getRetryDelay() {
+        return this.retryDelay;
+    }
+
+    /**
+     * @see RemoteAssetsConfig#getSaveInterval()
+     * @return Integer
+     */
+    @Override
+    public Integer getSaveInterval() {
+        return this.saveInterval;
+    }
+
+    /**
+     * @see RemoteAssetsConfig#getEventUserData()
+     * @return String
+     */
+    @Override
+    public String getEventUserData() {
+        return this.eventUserData;
+    }
+
+    /**
+     * @see RemoteAssetsConfig#getWhitelistedServiceUsers()
+     * @return String
+     */
+    @Override
+    public Set<String> getWhitelistedServiceUsers() {
+        return this.whitelistedServiceUsers;
+    }
+
+    /**
+     * @see RemoteAssetsConfig#getRemoteAssetsHttpExecutor()
+     * @return Executor
+     */
+    @Override
+    public Executor getRemoteAssetsHttpExecutor() {
+        return remoteAssetsHttpExecutor;
+    }
+
+    /**
+     * @see RemoteAssetsConfig#getResourceResolver()
+     */
+    @Override
+    public ResourceResolver getResourceResolver() {
+        try {
+            Map<String, Object> userParams = new HashMap<>();
+            userParams.put(ResourceResolverFactory.SUBSERVICE, RemoteAssets.SERVICE_NAME);
+            ResourceResolver resourceResolver = this.resourceResolverFactory.getServiceResourceResolver(userParams);
+            Session session = resourceResolver.adaptTo(Session.class);
+            if (StringUtils.isNotBlank(this.getEventUserData())) {
+                session.getWorkspace().getObservationManager().setUserData(this.getEventUserData());
+            }
+            return resourceResolver;
+        } catch (Exception e) {
+            LOG.error("Remote assets functionality cannot be enabled - service user login failed");
+            throw new RemoteAssetsServiceException(e);
+        }
+    }
+
+    /**
+     * @see RemoteAssetsConfig#closeResourceResolver(ResourceResolver)
+     */
+    @Override
+    public void closeResourceResolver(ResourceResolver resourceResolver) {
+        if (resourceResolver != null) {
+            try {
+                Session session = resourceResolver.adaptTo(Session.class);
+                if (session != null) {
+                    session.logout();
```

Comment:
No idea if this `logout()` is necessary - my gut says it isnt but I dont know enough of the internals to be sure.  The reason I'm concerned in the first place is b/c the `getResourceResolver()` function uses the session to `setUserData()` on the workspace observation manager so as to prevent triggering workflows.